tests/sentry/utils/test_committers.py

```diff
@@ -2,25 +2,160 @@
 
 from datetime import timedelta
 from django.utils import timezone
+from mock import Mock
+from uuid import uuid4
 
-from sentry.models import Release
+from sentry.models import Commit, CommitAuthor, CommitFileChange, Release, Repository
 from sentry.testutils import TestCase
-from sentry.utils.committers import get_previous_releases, score_path_match_length, tokenize_path
+from sentry.utils.committers import _get_commit_file_changes, _get_frame_paths, get_previous_releases, score_path_match_length, tokenize_path
 
 
-def test_score_path_match_length():
-    assert score_path_match_length('foo/bar/baz', 'foo/bar/baz') == 3
-    assert score_path_match_length('foo/bar/baz', 'bar/baz') == 2
-    assert score_path_match_length('foo/bar/baz', 'baz') == 1
-    assert score_path_match_length('foo/bar/baz', 'foo') == 0
-    assert score_path_match_length('./foo/bar/baz', 'foo/bar/baz') == 3
+class CommitTestCase(TestCase):
+    def setUp(self):
+        self.repo = Repository.objects.create(
+            organization_id=self.organization.id,
+            name=self.organization.id,
+        )
+
+    def create_commit(self, author=None):
+        return Commit.objects.create(
+            organization_id=self.organization.id,
+            repository_id=self.repo.id,
+            key=uuid4().hex,
+            author=author,
+        )
+
+    def create_commit_with_author(self, user=None, commit=None):
+        if not user:
+            user = self.create_user(name='Sentry', email='sentry@sentry.io')
+
+        author = CommitAuthor.objects.create(
+            organization_id=self.organization.id,
+            name=user.name,
+            email=user.email,
+            external_id=user.id,
+        )
+        if not commit:
+            commit = self.create_commit(author)
+        return commit
+
+    def create_commitfilechange(self, commit=None, filename=None, type=None):
+        return CommitFileChange.objects.create(
+            organization_id=self.organization.id,
+            commit=commit or self.create_commit(),
+            filename=filename or 'foo.bar',
+            type=type or 'M',
+        )
+
+
+class TokenizePathTestCase(TestCase):
+    def test_forward_slash(self):
+        assert list(tokenize_path('foo/bar')) == ['bar', 'foo']
+
+    def test_back_slash(self):
+        assert list(tokenize_path('foo\\bar')) == ['bar', 'foo']
+
+    def test_dot_does_not_separate(self):
+        assert list(tokenize_path('foo.bar')) == ['foo.bar']
+
+    def test_additional_slash_in_front(self):
+        assert list(tokenize_path('/foo/bar')) == ['bar', 'foo']
+        assert list(tokenize_path('\\foo\\bar')) == ['bar', 'foo']
+
+    def test_additional_slash_with_dot_in_front(self):
+        # TODO(lb): better name for this?
+        # TOOD(lb): Also.... I'm not sure we should be capturing '.'
+        assert list(tokenize_path('./')) == ['.']
+        assert list(tokenize_path('./../')) == ['..', '.']
```

Comment:
Commit paths shouldn't have `.` or `..` in them, but they could show up in stacktraces in theory. Given we won't have access to the rest of the file path capturing the dots is the best we can do.